### ISSUE TITLE
Ignored project files are also getting cached

### ISSUE THREAD

**Issue by kaushalmodi** [bbatsov/projectile#1024]
## Expected behavior

Files from ignored projects should not be cached.

## Actual behavior

Files from ignored projects are cached.

## Steps to reproduce the problem

Ignore some project and then open some file in that project. You will see a message in the minibuffer that that file is getting cached (even when the project is ignored).

### Projectile version information

```
Projectile version: 20160708.522
```

### Emacs version

```
Emacs version: GNU Emacs 25.1.50.16 (x86_64-unknown-linux-gnu, GTK+ Version 2.24.23)
 of 2016-07-10, built using commit 466ee1b3ea76425d201b5d59950e88251870c836.
```

### Operating system

RHEL 6.6

## My temporary solution

```el
(defun modi/advice-projectile-dont-cache-ignored-projects (&rest args)
  "Do not cache files from ignored projects when doing `find-file'."
  (let* ((project-root (projectile-project-p))
         (do-not-cache (projectile-ignored-project-p project-root)))
    do-not-cache))
(advice-add 'projectile-cache-files-find-file-hook :before-until
            #'modi/advice-projectile-dont-cache-ignored-projects)
```

**Comment by kaushalmodi** [comment]
https://github.com/bbatsov/projectile/pull/1025

**Comment by bbatsov** [comment]
Fix by #1025.
